presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ImplementIntersectAndExceptAsUnion.java

```diff
@@ -125,26 +129,52 @@ public PlanNode visitIntersect(IntersectNode node, RewriteContext<Void> rewriteC
             UnionNode union = union(withMarkers, ImmutableList.copyOf(concat(outputs, markers)));
 
             // add count aggregations and filter rows where any of the counts is >= 1
-            AggregationNode aggregation = computeCounts(union, outputs, markers);
-            FilterNode filterNode = addFilter(aggregation);
+            List<Symbol> aggregationOutputs = allocateSymbols(markers.size(), "count", BIGINT);
+            AggregationNode aggregation = computeCounts(union, outputs, markers, aggregationOutputs);
+            FilterNode filterNode = addFilterForIntersect(aggregation);
 
             return project(filterNode, outputs);
         }
 
-        private List<Symbol> allocateMarkers(int count)
+        @Override
+        public PlanNode visitExcept(ExceptNode node, RewriteContext<Void> rewriteContext)
+        {
+            List<PlanNode> sources = node.getSources().stream()
+                    .map(rewriteContext::rewrite)
+                    .collect(toList());
+
+            List<Symbol> markers = allocateSymbols(sources.size(), EXCEPT_MARKER, BOOLEAN);
+
+            // identity projection for all the fields in each of the sources plus marker columns
+            List<PlanNode> withMarkers = appendMarkers(markers, sources, node);
+
+            // add a union over all the rewritten sources. The outputs of the union have the same name as the
+            // original except node
+            List<Symbol> outputs = node.getOutputSymbols();
+            UnionNode union = union(withMarkers, ImmutableList.copyOf(concat(outputs, markers)));
+
+            // add count aggregations and filter rows where count for the first source is >= 1 and all others are 0
+            List<Symbol> aggregationOutputs = allocateSymbols(markers.size(), "count", BIGINT);
+            AggregationNode aggregation = computeCounts(union, outputs, markers, aggregationOutputs);
+            FilterNode filterNode = addFilterForExcept(aggregation, aggregationOutputs);
```

Comment:
is it possible to get the symbol list `aggregationOutputs` from `aggregation`? so then we need to pass only the `AggregationNode` to the method?